### ISSUE TITLE
[BUG] Problem with hyperparam DEPTH when trying to run HPO on CatBoost via AutoGluon

### ISSUE THREAD

**Issue by antoniotdelend** [autogluon/autogluon#4101]
**Bug Report Checklist**

- [x] I provided code that demonstrates a minimal reproducible example.  
- [x] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [x] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**
I can't run hyperparameter optimization for CatBoost if I define a search space for the depth parameter. If this parameter is not set or is assigned a fixed value, the optimization runs normally.

**Expected behavior**
Perform hyperparameter optimization with CatBoost as an algorithm, obtaining a leaderboard at the end and extracting the best parameters from the best trained model.

**To Reproduce**
# Setting space search for the model's parameters
cat_params = {'learning_rate': space.Real(0.0005, .05, log=True),
              'iterations': space.Int(lower=100, upper=300),
              'l2_leaf_reg': space.Real(1e-3, 100, log=True),
              'subsample': space.Real(0.8, 1, log=False),
              'rsm': space.Real(0.8, 1, log=False),
              'min_data_in_leaf': space.Int(lower=1, upper=100),
              "depth": space.Int(lower=5, upper=8, default=6),

              # fix params
              'boosting_type': 'Ordered',
              'loss_function': 'Logloss',
              'od_type': 'Iter',
              'od_wait': 20,
    }
hyperparameters = {'CAT': cat_params,}  # When these keys are missing from hyperparameters dict, no models of that type are trained

time_limit = 10*60  # train various models for ~2 min
num_trials = 50  # try at most 5 different hyperparameter configurations for each type of model
search_strategy = 'auto'  # to tune hyperparameters using random search routine with a local scheduler

hyperparameter_tune_kwargs = {  # HPO is not performed unless hyperparameter_tune_kwargs is specified
    'num_trials': num_trials,
    'scheduler' : 'local',
    'searcher': search_strategy,
}  # Refer to TabularPredictor.fit docstring for all valid values

save_path = 'agModels-predictClass_1'  # specifies folder to store trained models

tuned_predictor = TabularPredictor(label='target',
                            eval_metric='log_loss',
                            problem_type='binary',
                            path=save_path,
                            verbosity=0
                            ).fit(train_data=train, presets='best_quality', 
                                  time_limit=time_limit,
                                  hyperparameters=hyperparameters,
                                  hyperparameter_tune_kwargs=hyperparameter_tune_kwargs, 
                                  included_model_types=['CAT'],
                                  fit_weighted_ensemble=False,
                                  num_bag_folds=0,
                                  num_bag_sets=0,
                                  num_stack_levels=0,
                                  feature_generator=AutoMLPipelineFeatureGenerator(enable_text_special_features=False, 
                                                                                   enable_text_ngram_features=False)
)  

**Screenshots / Logs**
[bug error output.txt](https://github.com/autogluon/autogluon/files/14983163/bug.error.output.txt)
![image](https://github.com/autogluon/autogluon/assets/128329569/e49243df-d783-412a-bc74-0057ce5b29f1)

**Installed Versions**

```python

INSTALLED VERSIONS
------------------
date                : 2024-04-15
time                : 15:33:05.975379
python              : 3.10.5.final.0
OS                  : Windows
OS-release          : 10
Version             : 10.0.22631
machine             : AMD64
processor           : Intel64 Family 6 Model 140 Stepping 1, GenuineIntel
num_cores           : 8
cpu_ram_mb          : 16118.78515625
cuda version        : None
num_gpus            : 0
gpu_ram_mb          : []
avail_disk_size_mb  : None

async-timeout       : None
autogluon           : None
autogluon.common    : 1.0.0
autogluon.core      : 1.0.0
autogluon.features  : 1.0.0
autogluon.tabular   : 1.0.0
boto3               : 1.34.82
catboost            : 1.2.3
fastai              : None
hyperopt            : None
imodels             : None
lightgbm            : None
matplotlib          : 3.8.3
networkx            : 3.2.1
numpy               : 1.26.4
onnxruntime-gpu     : None
pandas              : 2.1.4
psutil              : 5.9.8
ray                 : None
requests            : 2.31.0
scikit-learn        : 1.4.0
scikit-learn-intelex: None
scipy               : 1.12.0
setuptools          : 69.2.0
skl2onnx            : None
tabpfn              : None
torch               : 2.0.1
tqdm                : 4.66.2
vowpalwabbit        : None
xgboost             : 2.0.3

```

</details>

**Comment by Innixma** [comment]
Thanks for the bug report @antoniotdelend! This bug has been fixed in the AutoGluon 1.1 release.
